src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -126,8 +126,14 @@ internal abstract class TexlFunction : IFunction
         /// </summary>
         public virtual bool HasPreciseErrors => false;
 
-        // Returns true if the function will mutate the value of argument 0, as is the case with Patch, Collect, Remove, etc.
-        public virtual bool MutatesArg0 => false;
+        /// <summary>
+        /// Returns true if the function will mutate the argument, as is the case of Patch, Collect, Remove, etc.
+        /// Set can also mutate, but needs to make a decision based on the argument's node.

```

Comment:
Can you give an example on the Set function where it "needs to make a decision based on the argument's node"?

Reply:
Good call, added.